src/utils/components/EmailInput.vue

```diff
@@ -0,0 +1,218 @@
+        default: null,
+    },
+    /**
+     * Valid message Message that will be added in green below the field once the validation has
+     * been done and the field is valid.
+     *
+     * @type {Sting}
+     */
+    validMessage: {
+        type: String,
+        default: '',
+    },
+    /**
+     * Mark the field as invalid
+     *
+     * This can be used if the field requires some external validation. When not set or set to null
+     * this props is ignored.
+     *
+     * NOTE: this props is ignored when activate-validation is false
+     *
+     * @type {Boolean}
+     */
+    invalidMarker: {
+        type: [Boolean, null],
+        default: null,
+    },
+    /**
+     * Invalid message Message that will be added in red below the field once the validation has
+     * been done and the field is invalid.
+     *
+     * NOTE: this message is overwritten if the internal validation failed (not allow file type or
+     * file too big or required empty file)
+     *
+     * @type {Sting}
+     */
+    invalidMessage: {
+        type: String,
+        default: '',
+    },
+    /**
+     * Mark the field has validated.
+     *
+     * As long as the flag is false, no validation is run and no validation marks are set. Also the
+     * props is-invalid and is-valid are ignored.
+     */
+    activateValidation: {
+        type: Boolean,
+        default: false,
+    },
+    /**
+     * Validate function to run when the input changes The function should return an object of type
+     * `{valid: Boolean, invalidMessage: Sting}`. The `invalidMessage` string should be a
+     * translation key.
+     *
+     * NOTE: this function is called each time the field is modified
+     *
+     * @type {Function | null}
+     */
+    validate: {
+        type: [Function, null],
+        default: null,
+        validator: propsValidator4ValidateFunc,
+    },
+    dataCy: {
+        type: String,
+        default: '',
+    },
+})
+const { placeholder, disabled, label, description } = toRefs(props)
+
+const { value, validMarker, invalidMarker, validMessage, invalidMessage, required, onFocus } =
+    useFieldValidation(props, model, emits, {
+        customValidate: validateEmail,
+        requiredInvalidMessage: 'no_email',
+    })
+
+const emailInputElement = ref(null)
+
+const dataCyPrefix = computed(() => (props.dataCy ? `${props.dataCy}-email-input` : `email-input`))
+
+function validateEmail() {
+    if (value.value && !isValidEmail(value.value)) {
+        return { valid: false, invalidMessage: 'invalid_email' }
+    }
+    return { valid: true, invalidMessage: '' }
+}
+
+function focus() {
+    emailInputElement.value.focus()
+}
+
+defineExpose({ focus })
+</script>
+
+<template>
+    <div class="form-group has-validation">
+        <label
+            v-if="label"
+            class="mb-2"
+            :class="{ 'fw-bolder': required }"
+            :for="inputEmailId"
+            :data-cy="`${dataCyPrefix}-label`"
```

Comment:
It might be better (more comprehensible) to add static/fixed `data-cy="..."` here, and then add another one whenever the component is used
```
<EmailInput data-cy="some-identifier" />
```
It will be applied to the div wrapping the label and input, and we can still access them in Cypress test with `cy.get('[data-cy="some-identifier"] [data-cy="..."]')`

Reply:
Done, hopefully it was easy with search and replace, will see if the test passes 🤞🏼